qutebrowser/keyinput/keyutils.py

```diff
@@ -52,6 +52,11 @@ def _is_printable(key):
     _assert_plain_key(key)
     return key <= 0xff and key not in [Qt.Key_Space, 0x0]
 
+# Unicode characters such as emoji and extended CJK characters land here
+# This checks for unicode codepoints above ASCII range. The highest unicode
+# is below Qt::Key modifiers and "special" keys, which start at 0x1000000
+def _is_extended_unicode(key):
+    return 0xff < key < 0x10ffff
```

Comment:
Shouldn't this be `< 0x1000000` or `<= 0x10ffff`? I'd prefer `< Qt.Key_Escape` though.

Reply:
Alternatively, to be absolutely sure we don't change the behavior of any existing code, it's possible to check if it's a character reserved for surrogates (`0xd800 <= key <= 0xdfff`)

Reply:
It should be `<=`, yes. And checking for the surrogates would be tighter, I'll run some tests to make sure that covers all the problem cases and probe the details of Qt's behavior.